velox/exec/ExchangeClient.cpp

```diff
@@ -117,6 +117,11 @@ ExchangeClient::next(uint32_t maxBytes, bool* atEnd, ContinueFuture* future) {
   std::vector<std::unique_ptr<SerializedPage>> pages;
   {
     std::lock_guard<std::mutex> l(queue_->mutex());
+    if (closed_) {
```

Comment:
It is good to have a check here but I think this can't totally prevent this from happening. We probably need closed_ flag inside the queue as queue close is not called with queue lock protection as exchange client close does.

Reply:
@xiaoxmeng 
ExchangeQueue by itself it not protected, instead we rely on protection in the ExchangeClient.
I believe this problem is fully solved.

We will observe the zombie tasks and if anything occurs - we will notice and address it.